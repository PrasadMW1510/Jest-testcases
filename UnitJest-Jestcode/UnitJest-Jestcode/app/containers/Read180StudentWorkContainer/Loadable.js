/**
 *
 * Asynchronously loads the component for Read180StudentWorkContainer
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
