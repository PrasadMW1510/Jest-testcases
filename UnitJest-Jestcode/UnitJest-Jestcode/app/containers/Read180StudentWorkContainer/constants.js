/*
 *
 * Read180StudentWorkContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/Read180StudentWorkContainer/DEFAULT_ACTION';
export const READ_180_STUDENT_WORK_LIST = 'app/Read180StudentWorkContainer/DEFAULT_ACTION';
export const READ_180_STUDENT_WORK_REQUEST =
  'containers/Read180NgContainer/Read180_Student_work_detail_request';
export const READ_180_STUDENT_WORK_REQUEST_SUCCESS =
  'containers/Read180NgContainer/Read180_Student_work_detail_request_success';
export const SET_READ_180_STUDENT_WORK_DATA =
  'containers/Read180NgContainer/set_read180_Student_work_data';
export const SET_180_STUDENT_WORK_REQUEST_SUCCESS =
  'containers/Read180NgContainer//set_read180_Student_work_data_success';
