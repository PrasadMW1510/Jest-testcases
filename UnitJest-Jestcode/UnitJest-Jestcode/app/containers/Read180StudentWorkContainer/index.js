/**
 *
 * Read180StudentWorkContainer
 *
 */

export { default } from './Read180StudentWorkContainer';
