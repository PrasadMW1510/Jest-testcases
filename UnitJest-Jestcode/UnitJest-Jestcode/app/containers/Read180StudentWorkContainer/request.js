import API, { getBaseUrlWithoutSlms } from 'utils/request';

export const getRead180StudentWorksAction = (read180Params, sessionId, userID) =>
  API.post(`${getBaseUrlWithoutSlms()}/slms/SlmsSdp`, read180Params, {
    params: {
      command: 'get_student_submissions',
      cohort_type: 'teacher',
      communityIds: 'R180NG,S44NG,S44JR,M180,M180Y2',
      sid: sessionId,
      cohort_id: userID,
    },
  }).then(response => response);

export const setRead180SwDataRequestAction = (read180Data, sessionId) =>
  API.post(`${getBaseUrlWithoutSlms()}/slms/SlmsSdp`, read180Data, {
    params: {
      command: 'evaluate_student_submissions',
      app_id: 'RTNG',
      sid: sessionId,
    },
  }).then(response => response);
