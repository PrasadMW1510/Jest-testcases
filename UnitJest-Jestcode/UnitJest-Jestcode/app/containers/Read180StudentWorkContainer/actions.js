/*
 *
 * Read180StudentWorkContainer actions
 *
 */

import * as Constants from './constants';

export function read180StudentWorkRequest(read180StudentWorkPreviewData) {
  return {
    type: Constants.READ_180_STUDENT_WORK_LIST,
    read180StudentWorkPreviewData,
  };
}
export function getRead180StudentWorkRequest(read180nData) {
  return {
    type: Constants.READ_180_STUDENT_WORK_REQUEST,
    read180nData,
  };
}

export function getRead180StudentWorkRequestSuccess(resultsData) {
  return {
    type: Constants.READ_180_STUDENT_WORK_REQUEST_SUCCESS,
    resultsData,
  };
}

export function setRead180StudentWorkData(read180ngData) {
  return {
    type: Constants.SET_READ_180_STUDENT_WORK_DATA,
    read180ngData,
  };
}

export function setRead180StudentWorkRequestSuccess(resultsData) {
  return {
    type: Constants.SET_180_STUDENT_WORK_REQUEST_SUCCESS,
    resultsData,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
