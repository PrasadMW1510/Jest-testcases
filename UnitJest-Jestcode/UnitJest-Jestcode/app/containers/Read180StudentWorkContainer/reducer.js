/*
 *
 * Read180StudentWorkContainer reducer
 *
 */

import { fromJS } from 'immutable';
import * as Constants from './constants';

const initialState = fromJS({});

function read180StudentWorkContainerReducer(state = initialState, action) {
  switch (action.type) {
    case Constants.READ_180_STUDENT_WORK_REQUEST_SUCCESS:
      return state
        .setIn(['read180Program', 'loading'], false)
        .setIn(['read180Program', 'itemCount'], fromJS(action.resultsData.item_count[0]))
        .setIn(['read180Program', 'results'], fromJS(action.resultsData.output_data[0]))
        .setIn(['read180Program', 'paginationData'], fromJS(action.resultsData.pagination_data[0]));
    case Constants.DEFAULT_ACTION:
      return state;
    default:
      return state;
  }
}

export default read180StudentWorkContainerReducer;
