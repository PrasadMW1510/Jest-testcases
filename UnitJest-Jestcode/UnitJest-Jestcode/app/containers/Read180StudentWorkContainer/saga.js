import { call, put, select, takeLatest } from 'redux-saga/effects';
import { showLoading, hideLoading } from 'containers/LoadingController/actions';
import { makeSelectProfileSessionId, makeSelectProfileUserId } from '../App/selectors';

import * as Constants from './constants';
import * as Request from './request';
import * as Actions from './actions';

export function* getRead180StudentWorkList(read180ninfo) {
  try {
    yield put(showLoading());
    const sessionId = yield select(makeSelectProfileSessionId());
    const userId = yield select(makeSelectProfileUserId());

    const read180Response = yield call(
      Request.getRead180StudentWorksAction,
      read180ninfo.read180nData,
      sessionId,
      userId
    );

    yield put(Actions.getRead180StudentWorkRequestSuccess(read180Response));
    yield put(hideLoading());
  } catch (err) {
    yield put(hideLoading());
  }
}

export function* setRead180StudentWorkData(postData) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());

    const read180Response = yield call(
      Request.setRead180SwDataRequestAction,
      postData.read180ngData,
      sessionId
    );
    yield put(Actions.setRead180StudentWorkRequestSuccess(read180Response));
  } catch (err) {
    yield put(hideLoading());
  }
}

export default function* defaultSaga() {
  yield takeLatest(Constants.READ_180_STUDENT_WORK_REQUEST, getRead180StudentWorkList);
  yield takeLatest(Constants.SET_READ_180_STUDENT_WORK_DATA, setRead180StudentWorkData);
}
