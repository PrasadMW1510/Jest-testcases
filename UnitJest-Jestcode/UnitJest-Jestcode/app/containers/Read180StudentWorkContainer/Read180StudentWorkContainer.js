/**
 *
 * Read180StudentWorkContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import Read180StudentWork from 'components/Read180StudentWork';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import { getRead180StudentWorkRequest, setRead180StudentWorkData } from './actions';

import makeSelectRead180StudentWorkContainer from './selectors';

import reducer from './reducer';
import saga from './saga';

export class Read180StudentWorkContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isFetch: false,
      moveForward: false,
      readStudentWork180Data: {},
      isEditable: true,
      selectedItemData: {},
    };
  }

  componentDidMount() {
    const { data } = this.props;
    if (data.row.communityId === 'RTNG') {
      let xmlinput = '';
      let studentId = data.row.id;
      if (data.row.studentId !== undefined) {
        studentId = data.row.studentId;
      }
      xmlinput = `<workItemsSubset>
    <workItemInfo 
    workItemId="${data.row.workItemId}"
     communityId="${data.row.communityId}" 
      studentId="${studentId}" 
      assignment="${data.row.assignment}" kind="${data.row.kind}" />
  </workItemsSubset>`;
      this.props.getRead180StudentWorkRequest(xmlinput);
    }
  }

  componentWillReceiveProps(nextProps) {
    const readStudentWork180Data =
      (nextProps.read180studentworkcontainer &&
        nextProps.read180studentworkcontainer.read180Program) ||
      {};
    if (!this.state.isFetch) {
      this.setState({
        isFetch: true,
        readStudentWork180Data,
      });
    }
    this.setState({
      readStudentWork180Data,
    });
  }

  savePostAssesment = (type, rubricType, comment, score) => {
    const postxmlInput = `<evaluationUpdate submissionType="${type}" studentId="${
      this.props.data.row.id
    }" teacherId="j107e7hc6rru0b9tql34mira_2efa7f0" rubricType="${rubricType}" workItemId="${
      this.props.data.row.workItemId
    }">
<questionNum>NaN</questionNum>
<comment>${comment}</comment>
<rubricScores>
  <score rubricOrder="${score}">
    <teacherScore>${score}</teacherScore>
  </score>
</rubricScores>
</evaluationUpdate>`;
    this.props.setRead180StudentWorkData(postxmlInput);
  };

  handlePreview = opts => {
    this.props.getRead180StudentWorkRequest(opts);
  };

  render() {
    return (
      <div>
        <meta name="description" content="Description of Read180StudentWorkContainer" />
        <Read180StudentWork
          readStudentWork180Data={this.state.readStudentWork180Data}
          savePostAssesment={this.savePostAssesment}
          {...this.props}
        />
      </div>
    );
  }
}

Read180StudentWorkContainer.propTypes = {
  getRead180StudentWorkRequest: PropTypes.func.isRequired,
  setRead180StudentWorkData: PropTypes.func.isRequired,
  read180studentworkcontainer: PropTypes.object.isRequired,
  data: PropTypes.object.isRequired,
};

const mapStateToProps = createStructuredSelector({
  read180studentworkcontainer: makeSelectRead180StudentWorkContainer(),
});

const withConnect = connect(mapStateToProps, {
  getRead180StudentWorkRequest,
  setRead180StudentWorkData,
});

const withReducer = injectReducer({ key: 'read180StudentWorkContainer', reducer });
const withSaga = injectSaga({ key: 'read180StudentWorkContainer', saga });

export default compose(withReducer, withSaga, withConnect)(Read180StudentWorkContainer);
