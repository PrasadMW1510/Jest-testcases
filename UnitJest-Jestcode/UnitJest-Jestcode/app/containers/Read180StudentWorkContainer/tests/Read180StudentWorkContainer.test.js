import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Read180StudentWorkContainer } from '../Read180StudentWorkContainer';

describe('Read180StudentWorkContainer  ', () => {
  let wrapper = null;
  let wrapper1 = null;
  let wrapper2 = null;

  const props = {
    handleCancel: jest.fn(),
    getRead180StudentWorkRequest: jest.fn(),
    setRead180StudentWorkData: jest.fn(),
    read180studentworkcontainer: {},
    data: {
      row: {
        communityId: 'RTNG',
        id: 'id',
        studentId: 'stud',
      },
    },
  };
  const props1 = {
    ...props,
    data: {
      row: {
        communityId: 'RTNG',
        id: 'id',
        studentId: undefined,
      },
    },
  };
  const props2 = {
    ...props,
    data: {
      row: {
        communityId: 'lll',
        id: 'id',
        studentId: undefined,
      },
    },
  };

  beforeEach(() => {
    wrapper = shallow(<Read180StudentWorkContainer {...props} />);
    wrapper1 = shallow(<Read180StudentWorkContainer {...props1} />);
    wrapper2 = shallow(<Read180StudentWorkContainer {...props2} />);
  });
  it('Should render Read180NgContainer when data is present ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(shallowToJson(wrapper1)).toMatchSnapshot();
    expect(shallowToJson(wrapper2)).toMatchSnapshot();
  });

  it('Should call componentWillReceiveProps ', () => {
    const nextProps = {
      read180studentworkcontainer: {
        read180Program: '',
      },
    };
    wrapper.instance().setState({
      isFetch: false,
    });
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state('isFetch')).toBeTruthy();
  });

  it('Should call componentWillReceiveProps ', () => {
    const nextProps = {
      read180studentworkcontainer: {
        read180Program: '',
      },
    };
    wrapper.instance().setState({
      isFetch: true,
    });
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state('isFetch')).toBeTruthy();
  });

  it('Should call handlePreview  ', () => {
    const opts = {};
    wrapper.instance().handlePreview(opts);
    expect(wrapper.instance().props.getRead180StudentWorkRequest).toMatchSnapshot();
  });

  it('Should call savePostAssesment ', () => {
    const type = '';
    const rubricType = '';
    const comment = '';
    const scoreead180Data = '';
    wrapper.instance().savePostAssesment(type, rubricType, comment, scoreead180Data);
    expect(wrapper.instance().props.setRead180StudentWorkData).toMatchSnapshot();
  });
});
