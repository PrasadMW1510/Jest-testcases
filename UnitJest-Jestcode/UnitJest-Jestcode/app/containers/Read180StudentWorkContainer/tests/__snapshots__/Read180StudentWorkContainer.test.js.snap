// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read180StudentWorkContainer   Should call handlePreview   1`] = `[Function]`;

exports[`Read180StudentWorkContainer   Should call savePostAssesment  1`] = `[Function]`;

exports[`Read180StudentWorkContainer   Should render Read180NgContainer when data is present  1`] = `
<div>
  <meta
    content="Description of Read180StudentWorkContainer"
    name="description"
  />
  <Read180StudentWork
    data={
      Object {
        "row": Object {
          "communityId": "RTNG",
          "id": "id",
          "studentId": "stud",
        },
      }
    }
    getRead180StudentWorkRequest={[Function]}
    handleCancel={[Function]}
    read180studentworkcontainer={Object {}}
    readStudentWork180Data={Object {}}
    savePostAssesment={[Function]}
    setRead180StudentWorkData={[Function]}
  />
</div>
`;

exports[`Read180StudentWorkContainer   Should render Read180NgContainer when data is present  2`] = `
<div>
  <meta
    content="Description of Read180StudentWorkContainer"
    name="description"
  />
  <Read180StudentWork
    data={
      Object {
        "row": Object {
          "communityId": "RTNG",
          "id": "id",
          "studentId": undefined,
        },
      }
    }
    getRead180StudentWorkRequest={[Function]}
    handleCancel={[Function]}
    read180studentworkcontainer={Object {}}
    readStudentWork180Data={Object {}}
    savePostAssesment={[Function]}
    setRead180StudentWorkData={[Function]}
  />
</div>
`;

exports[`Read180StudentWorkContainer   Should render Read180NgContainer when data is present  3`] = `
<div>
  <meta
    content="Description of Read180StudentWorkContainer"
    name="description"
  />
  <Read180StudentWork
    data={
      Object {
        "row": Object {
          "communityId": "lll",
          "id": "id",
          "studentId": undefined,
        },
      }
    }
    getRead180StudentWorkRequest={[Function]}
    handleCancel={[Function]}
    read180studentworkcontainer={Object {}}
    readStudentWork180Data={Object {}}
    savePostAssesment={[Function]}
    setRead180StudentWorkData={[Function]}
  />
</div>
`;
